[MCP] Added the update-record tool implementation.

anushakolan opened the pull request:
## Why make this change?

This PR,
- Adds the tool implementation for the `update-record` MCP tool in `Azure.DataApiBuilder.Mcp` to support updating records via MCP using keys and field payloads with full validation and authorization.
- Adds a README to guide contributors on how to test MCP tools using the MCP Inspector.
- Improves predicate construction logic in `SqlUpdateQueryStructure.cs` to support mapped column fields.

## What is this change?
- Adds `docs/Testing/mcp-inspector-testing.md` with step-by-step instructions for running MCP Inspector locally, bypassing TLS, and connecting to MCP endpoints.
- Introduces `UpdateRecordTool.cs` under BuiltInTools, enabling record updates via MCP with full validation, authorization, and mutation engine integration.
- Updates `SqlUpdateQueryStructure.cs` to construct predicates using parameterized values for safer and more flexible SQL generation.

Exceptions are thrown when:
1. Entity is null or empty.
2. Keys and Fields are not JSON objects.
3. Failed to parse Keys and Fields.
4. 0 keys provided.
5. 0 fields provided.
6. Key value is null or empty
7. field value is null or empty.

Error:
1. PermissionDenied - No permssions to execute.
2. InvalidArguments - No arguments provided.
3. InvalidArguments - Some missing arguments.
4. EntityNotFound - Entity not defined in the configuration.
5. InvalidArguments - No entity found with given key.
6. UnexpectedError - Any other UnexpectedError.

## How was this tested?

- [x] Manual testing via MCP Inspector
- [ ] Integration Tests
- [ ] Unit Tests

These scenarios were manually tested using MCP Inspector, as automated tests are not yet implemented.

### Valid Cases ###

1. Successful Update
- Provided valid entity, keys, and fields.
- Verified that values were correctly updated in the database.
- Confirmed that the response includes all the column values similar to REST API calls.

2. Permission Enforcement
- Modified role permissions and verified that access control is enforced correctly.

3. Composite Key Handling
- Tested entities requiring multiple keys.
- Verified correct update behavior and response formatting.

4. Alias Mapping
- Used mapped field name (e.g., order instead of position) and confirmed that only mapped names are accepted.

### Failure Cases ###


1. Empty Keys
- Provided an empty keys object.
- Received appropriate error indicating missing key values.

2. Empty Fields
- Provided an empty fields object.
- Received error indicating that at least one field is required.

3. Invalid Entity Name
- Used a non-existent entity name.
- Received EntityNotFound error.

4. Empty Entity String
- Provided an empty string for entity.
- Received InvalidArguments error.

5. Entity Not Configured
- Used an entity not present in the configuration (e.g., Todos).
- Received EntityNotFound error.

6. Null Entity
- Provided null for the entity field.
- Received InvalidArguments error.

8. Valid Key Format but Not in DB
- Used correct key structure but no matching record exists.
- Received error indicating no entity found with the given key.

9. Invalid or Null Key Values
- Provided null or malformed key values.
- Received InvalidArguments error.

10. Unauthorized Role Context
- Removed or misconfigured role header.
- Received PermissionDenied error.

## Sample Request(s)

`{ "id": "00000000-0000-0000-0000-000000000001" }`
`{ "title": "New Title", "order": 7 }`

anushakolan commented:
/azp run

anushakolan commented:
/azp run

Aniruddh25 commented on docs/Testing/mcp-inspector-testing.md:
```diff
@@ -0,0 +1,20 @@
+
+### 4. **How to use the tool**
+- Set the transport type "Streamable HTTP".
+- Set the URL "http://localhost:5000/mcp" and hit connect.
```
Mention as pre-requisite that dab mcp-server should be running before hitting connect here.

Aniruddh25 commented on docs/Testing/mcp-inspector-testing.md:
```diff
@@ -0,0 +1,20 @@
+set NODE_TLS_REJECT_UNAUTHORIZED=0
+
+### 3. ** Open the inspector with pre-filled token.**
+http://localhost:6274/?MCP_PROXY_AUTH_TOKEN=<token>
```
what is this token used for?

anushakolan commented on src/Azure.DataApiBuilder.Mcp/BuiltInTools/UpdateRecordTool.cs:
```diff
@@ -0,0 +1,471 @@
+
+            return new CallToolResult
+            {
+                Content =
```
Done, thanks for sharing this.

anushakolan commented:
/azp run

Aniruddh25 commented:
/azp run

anushakolan commented:
> LGTM. Thanks for addressing the comments and your patience. Please do create followup tasks for addressing the improvements.

Yes Ani, I will create the follow up tasks.
